commands/delete-registry.ts

```diff
@@ -0,0 +1,126 @@
+import * as vscode from "vscode";
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount, AzureSession } from '../typings/azure-account.api';
+import { RegistryRootNode } from "../explorer/models/registryRootNode";
+import { ServiceClientCredentials } from 'ms-rest';
+const teleCmdId: string = 'vscode-docker.deleteRegistry';
+import { AzureCredentialsManager } from '../utils/AzureCredentialsManager';
+import { AzureRegistryNode, AzureLoadingNode, AzureNotSignedInNode } from '../explorer/models/azureRegistryNodes';
+
+/**
+ * delete a registry and all it's associated nested items
+ * @param context : the AzureRegistryNode the user right clicked on to delete
+ */
+export async function deleteRegistry(context?: AzureRegistryNode) {
+    if (!context) {
+        deleteRegistryNoContext();
+        return;
+    }
+    else {
+        deleteRegistryContextAvailable(context);
+        return;
+    }
+}
+
+async function deleteRegistryContextAvailable(context: AzureRegistryNode) {
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: true,
+        placeHolder: 'No',
+        value: 'No',
+        prompt: 'Are you sure you want to delete this registry and its associated images? Enter Yes or No: '
+    };
+    let answer = await vscode.window.showInputBox(opt);
+    if (answer !== 'Yes') { return };
+
+    let azureAccount = context.azureAccount;
+    if (!azureAccount) {
+        return;
+    }
+
+    if (azureAccount.status === 'LoggedOut') {
+        return;
+    }
+
+    const registries = await AzureCredentialsManager.getInstance().getRegistries();
+    let resourceGroup: string;
+    for (let i = 0; i < registries.length; i++) {
+        if (registries[i].name === context.registry.name) {
+            resourceGroup = registries[i].id.slice(registries[i].id.search('resourceGroups/') + 'resourceGroups/'.length, registries[i].id.search('/providers/'));
+            break;
+        }
+    }
+    if (!resourceGroup) {
+        throw 'Something went wrong, this registry may no longer exist'
+    }
+
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(context.subscription);
+    client.registries.beginDeleteMethod(resourceGroup, context.registry.name).then(function (response) {
+        vscode.window.showInformationMessage('Successfully deleted registry ' + context.registry.name);
+    }, function (error) {
+        console.error("Failed!", error);
+        vscode.window.showErrorMessage(error);
+    })
```

Comment:
You should refactor the code and have a shared method to handle the deletion operation.